From 343b4ddfe03206ea9f3f37decc5ce2829e821bc9 Mon Sep 17 00:00:00 2001
Message-Id: <343b4ddfe03206ea9f3f37decc5ce2829e821bc9.1421830309.git.chang-joon.lee@intel.com>
In-Reply-To: <49dde4951ccd12ce81e60aed3b24793e0ac12fed.1421830308.git.chang-joon.lee@intel.com>
References: <49dde4951ccd12ce81e60aed3b24793e0ac12fed.1421830308.git.chang-joon.lee@intel.com>
From: Yogesh Mohan Marimuthu <yogesh.mohan.marimuthu@intel.com>
Date: Tue, 21 Oct 2014 23:02:29 +0530
Subject: [PATCH 08/17] REVERTME [VPG]: drm/i915: configure TE interrupt for
 CHV MIPI Port A

Configure TE interrupt for MIPI port A. Since TE interrupts for MIPI
port can be configured using multiple SoC pins, this code should be
handled by BIOS. currently hardcoding TE pin for MIPI port A as quick
solution.

REVERTME: This patch can be reverted once we have BIOS which will
configure TE or design in which it will be part of VBT configuration.

Issue: GMINL-1644
Change-Id: I3cf8bac129a2da4ed942896b8a028a7276f4ce34
Signed-off-by: Yogesh Mohan Marimuthu <yogesh.mohan.marimuthu@intel.com>
---
 drivers/gpu/drm/i915/i915_reg.h            |    7 +++++++
 drivers/gpu/drm/i915/intel_dsi_panel_vbt.c |   21 +++++++++++++++++++++
 2 files changed, 28 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_reg.h b/drivers/gpu/drm/i915/i915_reg.h
index 61c4c72..acf502d 100644
--- a/drivers/gpu/drm/i915/i915_reg.h
+++ b/drivers/gpu/drm/i915/i915_reg.h
@@ -539,8 +539,15 @@
 
 #define CHV_GPIO_CFG_UNLOCK			0x00000000
 #define CHV_GPIO_CFG_HiZ			0x00008100
+#define CHV_GPIO_CFG1_INT_RISING		0x00000002
+#define CHV_GPIO_CFG0_EDGE_DETECT		0x04000000
+#define CHV_GPIO_CFG0_5K			0x00100000
+#define CHV_GPIO_CFG0_TE_PAD			0x00030000
 #define CHV_GPIO_CFG_TX_STATE_SHIFT		1
 
+#define CHV_GPIO_NC_PNL1_BKLTCTL_CFG0		0x5418
+#define CHV_GPIO_NC_PNL1_BKLTCTL_CFG1		0x541c
+
 /* See configdb bunit SB addr map */
 #define BUNIT_REG_BISOC				0x11
 
diff --git a/drivers/gpu/drm/i915/intel_dsi_panel_vbt.c b/drivers/gpu/drm/i915/intel_dsi_panel_vbt.c
index 9ea6091..f5c27e7 100644
--- a/drivers/gpu/drm/i915/intel_dsi_panel_vbt.c
+++ b/drivers/gpu/drm/i915/intel_dsi_panel_vbt.c
@@ -964,6 +964,27 @@ static void generic_panel_reset(struct intel_dsi_device *dsi)
 	char *sequence = dev_priv->vbt.dsi.sequence[MIPI_SEQ_ASSERT_RESET];
 
 	generic_exec_sequence(intel_dsi, sequence);
+
+	if (is_cmd_mode(intel_dsi) && IS_CHERRYVIEW(dev)) {
+
+		/* this code should be in BIOS */
+
+		mutex_lock(&dev_priv->dpio_lock);
+
+		vlv_gpio_write(dev_priv, IOSF_PORT_GPIO_NC,
+			CHV_GPIO_NC_PNL1_BKLTCTL_CFG1, CHV_GPIO_CFG_UNLOCK);
+
+		vlv_gpio_write(dev_priv, IOSF_PORT_GPIO_NC,
+				CHV_GPIO_NC_PNL1_BKLTCTL_CFG1,
+				CHV_GPIO_CFG1_INT_RISING);
+
+		vlv_gpio_write(dev_priv, IOSF_PORT_GPIO_NC,
+				CHV_GPIO_NC_PNL1_BKLTCTL_CFG0,
+				CHV_GPIO_CFG0_EDGE_DETECT | CHV_GPIO_CFG0_5K |
+				CHV_GPIO_CFG0_TE_PAD);
+
+		mutex_unlock(&dev_priv->dpio_lock);
+	}
 }
 
 static void generic_disable_panel_power(struct intel_dsi_device *dsi)
-- 
1.7.9.5

